### ISSUE TITLE
LT function does not switch back from higher layer after button release for certain keys

### ISSUE THREAD

**Issue by brokenalarms** [jackhumbert/qmk_firmware#149]
For LT(LAYERX, KEYX), it works as expected for KEYX keys like KC_ENT, KC_UP, even KC_Z (thus not just non-alphanumeric), but for certain keys (haven't isolated it exactly, but so far it appears to be the home row and KC_SPCE), holding the key just switches to layer LAYERX and gets stuck there. This means that unless you have a separate toggle switch for that particular layer, you're just stuck there forever....

EDIT - I thought I'd just messed up, but after some more testing I still can't explain why LT(MDIA, KC_RGHT) and LT(MDIA, KC_A) both work fine but LT(MDIA, KC_SPC) and LT(MDIA, KC_U) both get stuck in the MDIA layer...

Thanks!

**Comment by brokenalarms** [comment]
My bad once reading instructions properly. Thanks :P

**Comment by mecanogrh** [comment]
Post your keymap nobody can tell what is exactly happening without seeing it.

**Comment by brokenalarms** [comment]
fixed, my issue. I didn't realize that held modifiers transforming to new layers need to be able to duplicate the original command on that new layer, either through KC_TRNS or repeating the original command. So my release events were being blocked by the assigned keys on the new layer.

Wouldn't have expected this behavior though...I'll rephrase and explain in new issue, thanks.
